src/macaron/slsa_analyzer/build_tool/base_build_tool.py

```diff
@@ -41,6 +42,14 @@ def file_exists(path: str, file_name: str) -> bool:
         return False
 
 
+@dataclass
+class RuntimeConfig:
+    """The class for build tool runtime configuration read from `defaults.ini`."""
+
+    #: The timeout used for running the build tool commands.
+    build_timeout: float = 600
```

Comment:
Could we just set an optional instance variable called `run_timeout` for `BaseBuildTool` sub-classes? Is this class going to store any other fields for each BuildTool?
Do you think that naming it `RunTimeConfig` would be confusing for developers as this class is entirely different from the "defaults.ini" config or the global config for example.

Reply:
> Could we just set an optional instance variable called `run_timeout` for `BaseBuildTool` sub-classes? Is this class going to store any other fields for each BuildTool?

I think we need to have a section in the `.ini` file for runtime settings explicitly. I expect more options to be added in future, which should be adjusted per build tool.

Reply:
> Do you think that naming it `RunTimeConfig` would be confusing for developers as this class is entirely different from the "defaults.ini" config or the global config for example.

Agreed. How about `RuntimeOptions`?

Reply:
See commit [fd880ec](https://github.com/oracle/macaron/pull/446/commits/fd880ec99dc041209783be5fc8980d7f0163966c)